protocol/v2/ssv/validator/committee_guard.go

```diff
@@ -34,6 +34,7 @@ func (a *CommitteeDutyGuard) StartDuty(role spectypes.BeaconRole, validator spec
 		return fmt.Errorf("unsupported role %d", role)
 	}
 	runningSlot, exists := duties[validator]
+	// TODO: leave a clarifying comment for this condition
 	if exists && runningSlot >= slot {
 		return fmt.Errorf("duty already running at slot %d", runningSlot)
 	}
```

Comment:
I've been puzzled by this, there is a [short discussion](https://github.com/ssvlabs/ssv/pull/1743/files#r1763483388) about this code line but it doesn't really answer **why it's OK to replace running duty with a new duty started afresh**. 

So I'd want to add a comment here if you could help me out, my default assumption is - we don't want to replace running duty (because nothing really prevents us from doing it over and over again) and rather prefer to finish the one started at the earliest time. But I'm clearly wrong on this ... perhaps this way we are trying to restart a duty that "got stuck" or something ?

Reply:
Due to our slashing protection mechanism, which would not let you sign an attestation older than the highest signed one. This means that if you sign on a newer duty, you will no longer be able to sign on the older one, therefore we prematurely stop the older to prevent this conflict and save resources.

There isn't an Ethereum slashing rule that strictly prevents you from signing an older attestation, but rather it's just a limitation of our slashing protection implementation, which can potentially be upgraded in the future.

Reply:
I see, that helps, I've added a clarification comment, thank you!

I guess my worry here was that we can continuously start new duties never letting the previous ones to finish. But I guess in practice there is always a large time-window between 2 duties for the same validator (at least 1 slot), which means it should never be a problem ?

Reply:
Right, updated the comment there.